src/razor/src/razorLanguageServerOptionsResolver.ts

```diff
@@ -35,23 +35,19 @@ export function resolveRazorLanguageServerOptions(
 }
 
 function findLanguageServerExecutable(withinDir: string) {
-    const extension = isWindows() ? '.exe' : '';
-    const executablePath = path.join(withinDir, `rzls${extension}`);
+    const isSelfContained = fs.existsSync(path.join(withinDir, 'coreclr.dll'));
     let fullPath = '';
-
-    if (fs.existsSync(executablePath)) {
-        fullPath = executablePath;
+    if (isSelfContained) {
+        const fileName = isWindows() ? 'rzls.exe' : 'rzls';
+        fullPath = path.join(withinDir, fileName);
     } else {
-        // Exe doesn't exist.
-        const dllPath = path.join(withinDir, 'rzls.dll');
-
-        if (!fs.existsSync(dllPath)) {
-            throw new Error(
-                vscode.l10n.t("Could not find Razor Language Server executable within directory '{0}'", withinDir)
-            );
-        }
+        fullPath = path.join(withinDir, 'rzls.dll');
```

Comment:
Just a note here - it is possible to publish a non-self contained executable if you wanted to
https://learn.microsoft.com/en-us/dotnet/core/deploying/#publish-framework-dependent

The benefit of the executable is that your process name is the executable name.  Whereas if you launch with `dotnet rzls.dll` your executable name will be 'dotnet'